src/Management/src/Endpoint/Services/ServiceDescriptorConverter.cs

```diff
@@ -0,0 +1,61 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileSystemGlobbing.Internal;
+using Steeltoe.Management.Endpoint.Hypermedia;
+
+namespace Steeltoe.Management.Endpoint.Services;
+internal class ServiceDescriptorConverter : JsonConverter<ServiceContextDescriptor>
+{
+    public override ServiceContextDescriptor Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        throw new NotImplementedException();
+    }
+
+    public override void Write(Utf8JsonWriter writer, ServiceContextDescriptor value, JsonSerializerOptions options)
+    {
+        writer.WriteStartObject();
+        writer.WritePropertyName("contexts");
+        writer.WriteStartObject();
+     
+        foreach (var contextKey in value.Keys)
+        {
+           
+            writer.WritePropertyName(contextKey);
+            writer.WriteStartObject();
+            writer.WritePropertyName("beans");
+            writer.WriteStartObject();
+            foreach (var (key, serviceDescriptor) in value[contextKey])
+            {
+                var name = serviceDescriptor.ServiceType.Name;
+             
+                writer.WritePropertyName(name);
+                writer.WriteStartObject();
+                writer.WriteString("scope", serviceDescriptor.Lifetime.ToString());
+                writer.WriteString("type", key);
+                writer.WriteString("resource", serviceDescriptor.ServiceType.AssemblyQualifiedName);
+                writer.WritePropertyName("dependencies");
+                
+                writer.WriteStartArray();
+                if (serviceDescriptor.ImplementationType != null)
+                {
+                    foreach (var constructor in serviceDescriptor.ImplementationType.GetConstructors())
```

Comment:
I agree with the first part, because it is used for disambiguation, but also very often at runtime, the constructor used depends on configuration - so we cannot be sure which one was used - unless we also query the service container to see if every type in the definition exists and if not go to the next best match. This adds complexity we can side step by simply listing all of them (and perhaps noting which constructor has this attribute on it).

Reply:
As discussed, changed this to return a list of all constructors since there is no guaranteed way of selecting which constructor would get invoked - it depends on user configuration or which extension method was used to wire them.

Reply:
I added the logic to check activator utils first.